src/main/scala/org/constellation/gossip/GossipService.scala

```diff
@@ -53,26 +54,37 @@ abstract class GossipService[F[_]: Parallel, A](
     }
 
     for {
-      _ <- messageTracker.remove(message.path.id)
-      failureNode <- bisectA(
+      _ <- logger.debug(s"Recovering path ${message.path.id}")
+      failureNode <- bisect.runA(
         (id: Id) => getClientMetadata(id).flatMap(validationFn(_, message)),
         sequence
       )
-      sequenceToRetry = sequence.dropWhile(node => !failureNode.contains(node))
-      clients <- sequenceToRetry.toList.traverse(getClientMetadata)
-      _ <- clients.parTraverseN(recoverThreadLimit) { client =>
-        generatePathId.flatMap { pathId =>
-          spreadFn(
-            client,
-            message
-              .copy(path = GossipPath(IndexedSeq(selfId, client.id, selfId), s"${message.path.id}_$pathId"))
-              .sign(keyPair)
-          )
-        }
-      } // TODO: Force trigger peer healthcheck if it fails
+      retryIds = sequence.dropWhile(node => !failureNode.contains(node))
+      _ <- retryIds.headOption.traverse { headId =>
+        F.start(C.shift >> retryHead(headId, message))
```

Comment:
The first one is actually needed. `head` is the first node on a path, which didn't get the message, so it's very likely that request to this node will either timeout, or just fail (connection error, or processing error). Therefore I think it's reasonable to run it independently from the main thread.

The second one is less needed because the `tail` is more likely to succeed, so I could actually remove the context shift there.